dashboard/app/controllers/data_docs_controller.rb

```diff
@@ -12,9 +12,24 @@ def create
 
     if @data_doc.save
       # TODO [meg] : Write serialization
-      render :ok, json: {} # TODO [meg] : Redirect to new data doc, e.g. redirect_to @data_doc
+      redirect_to action: :show, key: params[:key]
     else
       render :not_acceptable, json: @data_doc.errors
     end
   end
+
+  def to_param
+    key
+  end
+
+  # GET /data_docs/:key
+  def show
+    @data_doc = DataDoc.find_by(key: params[:key])
```

Comment:
I am curious, do you get this for free due to the combination of `load_and_authorize_resource` her and `param: :key` in routes.rb ? I would suggest removing this statement and seeing if everything still works (CanCan should define @data_doc for you). if not... no need to investigate too much, sometimes CanCan doesn't quite work as expected when the param is not `id`.

Reply:
Just tried it: @data_doc isn't defined ... also, in the `create` method, I originally tried
```
    if @data_doc.save
      redirect_to @data_doc
```
and got the same problem (it tried to use the id in the URL, which I was surprised by given the `param: :key` in routes)